test/ocpbugs48050/server.go

```diff
@@ -0,0 +1,217 @@
+		"0\r\n\r\n",
+	}
+
+	for i := range chunks {
+		if _, err := writer.WriteString(chunks[i]); err != nil {
+			return fmt.Errorf("error writing chunk: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// handleDuplicateTE handles the response for the /duplicate-te path.
+func handleDuplicateTE(writer *bufio.Writer) error {
+	response := fmt.Sprintf("HTTP/1.1 200 OK\r\n"+
+		"Date: %s\r\n"+
+		"Content-Type: text/plain; charset=utf-8\r\n"+
+		"Connection: close\r\n"+
+		"Transfer-Encoding: chunked\r\n"+
+		"Transfer-Encoding: chunked\r\n\r\n", time.Now().UTC().Format(time.RFC1123))
+
+	if _, err := writer.WriteString(response); err != nil {
+		return fmt.Errorf("error writing duplicate-te response headers: %v", err)
+	}
+
+	chunks := []string{
+		// First chunk:
+		// - 'D' is the hexadecimal representation of 13 (the length of "duplicate-te\n").
+		// - \r\n separates the chunk size from the chunk data.
+		// - "duplicate-te\n" is the chunk data (note the \n is part of the data).
+		// - \r\n terminates the chunk.
+		"D\r\nduplicate-te\n\r\n",
+
+		// Final chunk:
+		// - '0' indicates a chunk of zero length, signaling the end of the chunked message.
+		// - The first \r\n separates the chunk size (0) from what would be the chunk data.
+		// - The second \r\n terminates the zero-length chunk and the entire chunked message.
+		"0\r\n\r\n",
+	}
+
+	for i := range chunks {
+		if _, err := writer.WriteString(chunks[i]); err != nil {
+			return fmt.Errorf("error writing chunk: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func createListener(port string, useTLS bool) (net.Listener, error) {
+	if useTLS {
+		cert, err := tls.LoadX509KeyPair("/etc/serving-cert/tls.crt", "/etc/serving-cert/tls.key")
+		if err != nil {
+			return nil, fmt.Errorf("error loading TLS certificate: %w", err)
+		}
+
+		config := &tls.Config{
+			Certificates: []tls.Certificate{cert},
+		}
+		return tls.Listen("tcp", ":"+port, config)
+	}
+
+	return net.Listen("tcp", ":"+port)
+}
+
+func startServer(port string, useTLS bool) {
+	listener, err := createListener(port, useTLS)
+	if err != nil {
+		log.Fatalf("Error starting server on port %s: %v\n", port, err)
+	}
+
+	defer listener.Close()
+
+	fmt.Printf("Server is listening on port %v (useTLS=%v)\n", port, useTLS)
+
+	for {
+		conn, err := listener.Accept()
+		if err != nil {
+			fmt.Println("Error accepting connection:", err)
+			continue
+		}
+		go handleConnection(conn)
+	}
+}
+
+func Serve() {
+	httpPort := os.Getenv("HTTP_PORT")
+	httpsPort := os.Getenv("HTTPS_PORT")
+
+	if httpPort == "" || httpsPort == "" {
+		log.Fatalf("Environment variables HTTP_PORT and HTTPS_PORT must be set\n")
+	}
+
+	// Start non-TLS server handling all paths, including /healthz.
+	go startServer(httpPort, false)
+
+	// Start TLS server handling all paths, including /healthz.
+	go startServer(httpsPort, true)
+
+	// Block forever (needed to keep the servers running).
+	select {}
```

Comment:
How to terminate this command if I run it in my bash? Should you do a signal handling to react to Ctrl-C at similar signals?

Reply:
Why? This server only runs for the duration of the test.

Reply:
> How to terminate this command if I run it in my bash? Should you do a signal handling to react to Ctrl-C at similar signals?

Hit Ctrl-C. Or Ctrl-\. If you're running it locally, both will terminate the command.